process/src/services/service.ports.ts

```diff
@@ -1,295 +1,379 @@
 
-        this._controllers.forEach((controller: ControllerSerialPort) => {
-            this._connectedPortsState.state[controller.getPath()] = {
-                connections: this._getConnectionsCount(controller.getPath()),
-                ioState: controller.getIOState()
-            };
-            controller.clearIOState();
-        });
-        if (Object.keys(this._connectedPortsState.state).length === 0) {
-            return;
-        }
-        if (this._connectedPortsState.attempts < 10) {
-            this._connectedPortsState.attempts += 1;
-            this._connectedPortsState.timer = setTimeout(() => {
-                this._sendConnectedPortsState();
-            }, 250);
-        } else {
-            this._sendConnectedPortsState();
-        }
+    this._controllers.forEach((controller: ControllerSerialPort) => {
+      this._connectedPortsState.state[controller.getPath()] = {
+        connections: this._getConnectionsCount(controller.getPath()),
+        ioState: controller.getIOState(),
+      };
+      controller.clearIOState();
+    });
+    if (Object.keys(this._connectedPortsState.state).length === 0) {
+      return;
     }
-
-    private _updateControllerSignatureState(session: string) {
-        const ports: string[] = [];
-        this._listeners.forEach((listeners: Map<string, IListeners>, port: string) => {
-            if (listeners.has(session)) {
-                ports.push(port);
-            }
-        });
-        ports.forEach((port: string) => {
-            let controller: ControllerSerialPort | undefined = this._controllers.get(port);
-            if (controller === undefined) {
-                return;
-            }
-        });
+    if (this._connectedPortsState.attempts < 10) {
+      this._connectedPortsState.attempts += 1;
+      this._connectedPortsState.timer = setTimeout(() => {
+        this._sendConnectedPortsState();
+      }, 250);
+    } else {
+      this._sendConnectedPortsState();
     }
+  }
 
-    private _sendConnectedPortsState() {
-        if (typeof this._token !== 'string') {
-            return;
+  private _updateControllerSignatureState(session: string) {
+    const ports: string[] = [];
+    this._listeners.forEach(
+      (listeners: Map<string, IListeners>, port: string) => {
+        if (listeners.has(session)) {
+          ports.push(port);
         }
-        this._connectedPortsState.attempts = 0;
-        PluginIPCService.sendToPluginHost('*', {
-            event: ERenderEvents.state,
-            streamId: '*',
-            token: this._token,
-            state: this._connectedPortsState.state
-        });
+      }
+    );
+    ports.forEach((port: string) => {
+      let controller: ControllerSerialPort | undefined =
+        this._controllers.get(port);
+      if (controller === undefined) {
+        return;
+      }
+    });
+  }
+
+  private _sendConnectedPortsState() {
+    if (typeof this._token !== "string") {
+      return;
```

Comment:
Shell we put here some logs? Like `warn` ~ cannot send port's state, because of token problems.... or something like it